lib/semantics/check-select-stmt.cc

```diff
@@ -0,0 +1,135 @@
+//==check-select-stmt.cc - Checker for select-rank
+// TODO:select-case
+// TODO:select-type
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===--------------------------------------------------------------------------------------===
+
+#include "check-select-stmt.h"
+#include "tools.h"
+#include "../common/Fortran.h"
+#include "../common/idioms.h"
+#include "../parser/message.h"
+#include "../parser/tools.h"
+#include <list>
+#include <optional>
+#include <set>
+#include <tuple>
+#include <variant>
+
+namespace Fortran::semantics {
+
+void SelectStmtChecker::Leave(
+    const parser::SelectRankConstruct &selectRankConstruct) {
+  const auto &selectRankStmt{
+      std::get<parser::Statement<parser::SelectRankStmt>>(
+          selectRankConstruct.t)};
+  const auto &selectRankStmtSel{
+      std::get<parser::Selector>(selectRankStmt.statement.t)};
+  // R1149 select-rank-stmt checks
+  std::visit(
+      common::visitors{
+          [](const parser::Expr &) {
+            // TODO:1.Checks if selector is an
+            // expression ex.Rank(a+b)
+          },
+          [&](const parser::Variable &variable) {
+            if (const Symbol * entity{GetLastName(variable).symbol}) {
+              if (!IsAssumedRankArray(*entity)) {  // C1150
+                context_.Say(variable.GetSource(),
+                    "Selector is not an assumed-rank array variable"_err_en_US);
+              }
+            }
+          },
+          [](auto &) {
+            common::die("Illegal value of selector in SELECT RANK "
+                        "statement");
+          },
```

Comment:
You don't need this lambda: `Expr` and `Variable` are the only alternatives in `Selector`.

Reply:
Thank you @tskeith I have removed it.
When I run `make test ` on local
The following tests FAILED:
        **`191 - call12.f90 (Failed)`**
Whereas committing a change upstream the drone.ci bots show results as passing.
Why does that happen, am I missing something while testing them on local?

Reply:
I doubt this change is causing a test to fail. If you put the code back it does it pass?